tests/get-coverage.yml

```diff
@@ -0,0 +1,73 @@
+# SPDX-License-Identifier: BSD-3-Clause
+---
+- name: Prepare for coverage extraction
+  hosts: all
+  tasks:
+    - name: set facts
+      set_fact:
+        coverage_module: network_connections
+        coverage: /root/.local/bin/coverage
+    - name: set more facts
+      set_fact:
+        coverage_file: ansible-coverage-{{ coverage_module }}-{{ test_playbook|replace('.yml', '') }}
+
```

Comment:
Can't you just use `vars:` at the top of the play?

Reply:
The problem with using `vars:` is that the variables are only set for the current playbook but I need it in several playbooks:

```
---
- name: set var
  hosts: all
  vars:
    - example_var: foo

- name: Show the var
  hosts: all
  tasks:
    - debug:
        var: example_var
```

```
ansible-playbook -i rhel7, test_things.yml

PLAY [set var] ****************************************************************************************************************************************************************************************************

TASK [Gathering Facts] ********************************************************************************************************************************************************************************************
ok: [rhel7]

PLAY [Show the var] ***********************************************************************************************************************************************************************************************

TASK [Gathering Facts] ********************************************************************************************************************************************************************************************
ok: [rhel7]

TASK [debug] ******************************************************************************************************************************************************************************************************
ok: [rhel7] => {
    "example_var": "VARIABLE IS NOT DEFINED!"
}

PLAY RECAP ********************************************************************************************************************************************************************************************************
rhel7                      : ok=3    changed=0    unreachable=0    failed=0
```

Reply:
can you please add a comment explaining whi it needs to be this way? Also, why have two `set_facts:` ?

Reply:
Done